native_client/deepspeech.cc

```diff
@@ -105,13 +112,24 @@ Model::infer(float* aMfcc, int aNFrames, int aFrameLen)
   // Output is an array of shape (1, n_results, result_length).
   // In this case, n_results is also equal to 1.
   auto output_mapped = outputs[0].tensor<int64, 3>();
-  int length = output_mapped.dimension(2) + 1;
-  char* output = (char*)malloc(sizeof(char) * length);
-  for (int i = 0; i < length - 1; i++) {
+  int output_length = output_mapped.dimension(2) + 1;
+
+  int decoded_length = 1; // add 1 for the \0
+  for (int i = 0; i < output_length - 1; i++) {
+    int64 character = output_mapped(0, 0, i);
+    const std::string& str = mPriv->alphabet->StringFromLabel(character);
+    decoded_length += str.size();
+  }
+
+  char* output = (char*)malloc(sizeof(char) * decoded_length);
+  char* pen = output;
+  for (int i = 0; i < decoded_length - 1; i++) {
```

Comment:
Shouldn't the iteration be done against `output_length` and not `decoded_length` here?

Reply:
Yep! Fixed.